test/siphash_internal_test.c

```diff
@@ -257,7 +257,7 @@ static int test_siphash(int idx)
 static int test_siphash_basic(void)
 {
     SIPHASH siphash = { 0, };
-    unsigned char key[SIPHASH_KEY_SIZE];
+    unsigned char key[SIPHASH_KEY_SIZE] = {0};
```

Comment:
This isn't needed, its initalized on line 191 before use

Reply:
There , it is initialized in the function test_siphash() but not in test_siphash_basic(). In both the functions, it's a local variable . In test_siphash_basic(), it's being passed directly to the following function SipHash_Init().

TEST_true(SipHash_Init(&siphash, key, 0, 0));

Reply:
Also, test_siphash_basic() is called at the earliest as suggested by the following code:

int setup_tests(void)
{
    ADD_TEST(test_siphash_basic);
    ADD_ALL_TESTS(test_siphash, OSSL_NELEM(tests));
    return 1;
}

Reply:
in test_siphash_basic, it seems whatever garbage is on the stack is being used as a key, and SipHash_Init translates the unsigned char that we pass to two uint64 variables starting at indexes 0 and 8.  Given that SIPHASH_KEY_SIZE is defined as 16, it won't overrun the array.  It doesn't hurt to initalize it, but what you're doing means we're going to test with a key that is all zeros, which is fine, but probably not whats intended.  If it needs to be initalized, it should likely be initalized with some concrete key that is something other than all zeros (though that might not hurt to test for either)

Reply:
Uninitialised memory on the stack will either lead to unspecified values or trap representations being used. A trap representation would lead to undefined behaviour, but `uint64_t` on two's complement systems (in practice all that are likely to be seen) does not have a trap value, so on two's complement systems this won't lead to undefined behaviour.

On one's complement systems it's implementation-defined whether "all ones" is a trap representation, so this could potentially be an issue there.

I agree that "all zeroes" might not be the best value to use - it might be best to bring in the initialisation from `test_siphash()`

Reply:
I think using all-zero key for this particular test is OK. Please just make the variable static const as well.

Reply:
Done. Please review.